### ISSUE TITLE
Wire up steam events

### ISSUE THREAD

**Issue by Licho1** [ZeroK-RTS/Chobby#109]
Steam integration todo

 - [x] SteamOnline callin sends you AuthToken, pass this to register or login message, you can also call extra LinkSteam if you are already connect and receive this event.

- [x] It sends you friend list on SteamOnline, use this to request friendship of those that arent friends, values are steamid, if someone is your friend you should see their steamid (logic can be changed)

- [x] It sends you SteamJoinFriend with steamID to request joining of friends game.
Logic should be: join same lobby room, if MM active, join party 

- [x] FriendID can also be sent in SteamOnline if invitation was made for offline user

- [x] There is SteamInviteFriend callout where you can invite friends (even offline if you know id)

- [x] There is SteamOpenOverlaySection - call it to show list of steam friends to invite to game

- [ ] There is SteamOpenOverlayWebsite - open URL in overlay (or browser if overlay not available)

- [x] SteamOverlayChanged callin is called when overlay opens/closes, use this to set higher fps 

- [x] Make use of MySteamName (sanitized)

**Comment by GoogleFrog** [comment]
-- WARNING: FPS needs to be increased, overlay works at chobby FPS

I'll need to know when the overlay should be closed. Also steam can open the overlay by itself so I need to know whenever it opens.

**Comment by GoogleFrog** [comment]
The list of friend IDs has limited usefulness if the users with corresponding steamID are not visible. A list of usernames would be instantly useful. Otherwise the server has to ensure that steam friends are visible, which is more work than finding them and sending their usernames.

**Comment by GoogleFrog** [comment]
I don't know how to see friend matchmaker or party status.

**Comment by Licho1** [comment]
Wrapper does not know/cannot know username without connecting to lobby server, which is task for the chobby.
I think we should:
1) allow friend by steamid
2) make callins for overlay on/off
3) list steamid for all friends

**Comment by GoogleFrog** [comment]
See https://github.com/ZeroK-RTS/Chobby/commit/e2686a47646f83034b6d55b7c919c767f584414c

**Comment by Licho1** [comment]
1,2,3 are done, wire up overlay event and new steamid parameter for friends

**Comment by GoogleFrog** [comment]
" It sends you friend list on SteamOnline, use this to request friendship of those that arent friends, values are steamid, if someone is your friend you should see their steamid (logic can be changed)"

How do I become ZKLS::friends with someone if they can only see the steamID of ZKLS::friends and I need to pick out my steam::friends only by their steamID?

**Comment by Licho1** [comment]
Friend lists and befriending now support steamid so you check existing friendlist and if not there you request it.

**Comment by Licho1** [comment]
You can test stuff against test server

**Comment by GoogleFrog** [comment]
I don't know what it means when you say they support steamID.

**Comment by Licho1** [comment]
Look at messages.cs

**Comment by GoogleFrog** [comment]
I think everything except overlay interaction is complete.

**Comment by GoogleFrog** [comment]
I'll do overlay when the install isn't 6GB.

**Comment by Licho1** [comment]
Ultimatum

**Comment by GoogleFrog** [comment]
No, I thought 6GB is the agreed upon plan and will be done sometime. Overlay can wait, I just don't want to wait for 6GB to download if that's the only think I need it for.

**Comment by Licho1** [comment]
ok, done download reduced to 1.2gb so go ahead.

**Comment by Licho1** [comment]
I think i can force overlay refresh from wrapper too if needed.

**Comment by Licho1** [comment]
Use of steam browser not enabled. This is more graceful than minimizing lobby to open chrome

**Comment by Licho1** [comment]
Reners site terribly? What do you mean? We can change site but i havent seen it in steam yet.

**Comment by The-Yak** [comment]
https://i.imgur.com/5gl9zK0.jpg

The site is just a little too wide to display well in default scaled steam browser. 1920x1080

**Comment by Licho1** [comment]
So what should be done? Improve site or use external browser?

**Comment by The-Yak** [comment]
IMO the former. Steam browser already works quite well for some pages like Commander editing and map browsing. Planetwars page is the worst in steam browser, but I am already hoping that the lovely campaign interface is used for planetwars instead.
